Sluggo/View Controllers/LoginViewController.swift

```diff
@@ -72,6 +72,24 @@ class LoginViewController: UIViewController {
                 // Save to identity
                 self.identity.token = record.key
                 
+                // Wait for user record to also be fetched
+                // Don't ask me why this needs to be in another dispatch, it just fails otherwise.
+                DispatchQueue.global(qos: .userInitiated).sync {
+                    userManager.getRecord(identity: self.identity) { result in
+                        switch(result) {
+                        case .success(let userRecord):
+                            self.identity.authenticatedUser = userRecord
+                            break
+                        case .failure(let error):
+                            print("FAILURE!")
+                            DispatchQueue.main.sync {
+                                UIAlertController.createAndPresentError(vc: self, error: error, completion: nil)
+                            }
+                        }
+                    }
+                    if(self.identity.authenticatedUser == nil) { return }; // avoid segueing out if we didn't fully successfully log in
```

Comment:
Since getRecord is asynchronous, will the authenticatedUser check always fail whenever the request does not complete before the check is done? Seems like whatever controls the segue should just go within this completion handler